hudi-flink-datasource/hudi-flink/src/main/java/org/apache/hudi/source/stats/FileStatsIndex.java

```diff
@@ -103,17 +105,22 @@ public String getIndexPartitionName() {
   public HoodieTableMetadata getMetadataTable() {
     // initialize the metadata table lazily
     if (this.metadataTable == null) {
-      HoodieHadoopStorage storage = new HoodieHadoopStorage(basePath, FlinkClientUtil.getHadoopConf());
-      HoodieTableMetaClient metaClient = HoodieTableMetaClient.builder().setBasePath(basePath).setConf(storage.getConf()).build();
+      initMetaClient();
       this.metadataTable = metaClient.getTableFormat().getMetadataFactory().create(
           HoodieFlinkEngineContext.DEFAULT,
-          storage,
-          metadataConfig,
+          metaClient.getStorage(),
+          StreamerUtil.metadataConfig(conf),
           basePath);
     }
     return this.metadataTable;
   }
 
+  private void initMetaClient() {
+    if (this.metaClient == null) {
+      this.metaClient = StreamerUtil.createMetaClient(conf);
```

Comment:
@danny0405 general question for you. is `StreamerUtil` the right place for such reusable helper methods

Reply:
yeah, for most of the the utilities method we put it here. we also have other utilities, you can check all the other utility classes under the same package.